### PR TITLE
Delete table from build_rules.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Delete table for test.

### CODE REVIEW

File: docs/reference/build_rules.md [hunk 1]
```diff
@@ -8,18 +8,6 @@ The build tools include every source file found in the project unless it is in a
 
 Mbed OS build tools determine the type of source file found based on the file's extension. The following table lists all files used in a build:
```
Comment (iriark01) [inline docs/reference/build_rules.md:9]: I can leave a comment

File: docs/reference/build_rules.md [hunk 2]
```diff
@@ -8,18 +8,6 @@ The build tools include every source file found in the project unless it is in a
...
-| `.cc`, `.cpp`       | C++ source        | Toolchain's C++ compiler       |
-| `.h`, `.hpp`, `.hh` | C/C++ header      | Toolchain's C and C++ compiler |
-| `.icf`              | IAR linker file   | IAR's linker                   |
-| `.ld`               | GCC linker script | GCC's linker (`ld`)            |
-| `.o`                | Object            | Toolchain's linker             |
-| `.s`, `.S`          | Assembly          | Toolchain's assembler          |
-| `.sct`              | ARM scatter file  | ARM's linker (`armlink`)       |
-
 
 ### Label directories
```
Comment (iriark01) [inline docs/reference/build_rules.md:12]: I'll add another comment to be annoying

**Review by iriark01 — CHANGES_REQUESTED** [review]
I left a few comments.

File: docs/reference/build_rules.md [hunk 3]
```diff
@@ -8,18 +8,6 @@ The build tools include every source file found in the project unless it is in a
 
 Mbed OS build tools determine the type of source file found based on the file's extension. The following table lists all files used in a build:
```
Comment (iriark01) [inline docs/reference/build_rules.md:10]: This will email Amanda

Comment (iriark01) [inline docs/reference/build_rules.md:10]: Won't email her yet

**Review by iriark01 — CHANGES_REQUESTED** [review]

**Review by iriark01 — APPROVED** [review]
This is cool

**Comment by AnotherButler** [comment]
Closing because this was an example in a demo

**Pull request closed without merging**